### PR TITLE
ci: add github workflow

### PR DESCRIPTION
**Pull request by x1ddos**
An alternative to Travis CI.

### CODE REVIEW

**Comment by x1ddos** [comment]
@benma quite a few annotations but mostly harmless. Click on the "Files changed" tab and scroll down.

**Comment by benma** [comment]
https://github.com/digitalbitbox/bitbox02-api-go/pull/40/checks?check_run_id=746039623
```
 Running [/home/runner/golangci-lint-1.26.0-linux-amd64/golangci-lint run --out-format=github-actions --skip-dirs=cmd/playground --issues-exit-code=0] in [] ...
  ##[error]Top level comment should end in a period (godot)
  ##[error]Top level comment should end in a period (godot)
  ##[error]Top level comment should end in a period (godot)
  ##[error]`if ok` is deeply nested (complexity: 9) (nestif)
  ##[error]err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"fail\")" (goerr113)
  ##[error]err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"fail\")" (goerr113)
  ##[error]err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"unexpected query\")" (goerr113)
  ##[error]err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"2 <= m <= n <= 15 must hold (m = threshold, n = number of signers)\")" (goerr113)
  ##[error]err113: do not define dynamic errors, use wrapped static errors instead: "fmt.Errorf(\"name must be %d chars or less\", multisigNameMaxLen)" (goerr113)
  ##[error]err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"error\")" (goerr113)
  ##[error]err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"error\")" (goerr113)
  ##[error]err113: do not define dynamic errors, use wrapped static errors instead: "errors.New(\"error\")" (goerr113)
  ##[error]mnd: Magic number: 3, in <condition> detected (gomnd)
  ##[error]mnd: Magic number: 0xee, in <argument> detected (gomnd)
  ##[error]mnd: Magic number: 0xFFFF, in <argument> detected (gomnd)
  ##[error]mnd: Magic number: 7, in <condition> detected (gomnd)
  ##[error]mnd: Magic number: 5, in <condition> detected (gomnd)
```

what's with all the errors? they don't show in travis.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  lint:
+    runs-on: ubuntu-18.04
+    steps:
+      - name: Clone the repo
+        uses: actions/checkout@v2
+      # https://github.com/golangci/golangci-lint-action
+      - name: Lint
+        uses: golangci/golangci-lint-action@v1
+        with:
+          version: v1.26
```
Comment (benma) [inline .github/workflows/ci.yml:16]: travis uses v1.21.0.

maybe update travis too, and fix the lints by disabling the newly added default inters in `.golangci.yml` (some of the new linters are really stupid lol!), or downgrade this for now?

**Comment by x1ddos** [comment]
Yeah, I've just installed v1.27.0 with:

     GO111MODULE=on go get -v 
github.com/golangci/golangci-lint/cmd/golangci lint@v1.27.0

and can reproduce same errors locally. Will try to disable stupid 
linters first, in the hopes they may have improved heuristics in the 
others we actually use.

So, I'll try to stay at 1.27. We'll see.

**Comment by benma** [comment]
@x1ddos please finish this PR - I think Travis is gone from this repo too, and we want at least one CI running on the PRs.

**Comment by x1ddos** [comment]
Let's see if this still works. I'll probably need to force-push a few times.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (x1ddos) [reply .github/workflows/ci.yml]: done. updated to v1.27.0, same as in the bb app repo.

**Comment by x1ddos** [comment]
PTAL

File: .golangci.yml [hunk 2]
```diff
@@ -65,35 +65,40 @@ linters-settings:
...
 
 linters:
   enable-all: true
   disable:
-    - maligned
-    - prealloc
-    - lll
-    - gochecknoinits
+    - gosec
+    - bodyclose
```
Comment (benma) [inline .golangci.yml:84]: was that producing some error? in this lib we don't have a server.

File: .golangci.yml [hunk 3]
```diff
@@ -102,12 +107,17 @@ issues:
...
+
+  # Excluding configuration per-path, per-linter, per-text and per-source.
+  exclude-rules:
+    # In addition to always disabled linters above, exclude some more in tests.
+    - path: _test\.go
+      linters:
+        - errcheck
+        - gocyclo
+        - godot
+        - gosec
```
Comment (benma) [inline .golangci.yml:122]: already included above.

**Review by benma — APPROVED** [review]
nice, thanks

File: .golangci.yml (hunk 2 above)
Comment (x1ddos) [reply .golangci.yml]: no. just copied it over from the bb app repo. removed.

**Pull request merged**